apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/index.tsx

```diff
@@ -133,18 +178,89 @@ const SyncOnboardingManual = () => {
     }
   }, [device]);
 
+  useEffect(() => {
+    if (deviceOnboardingState?.isOnboarded) {
+      setStepKey(StepKey.SoftwareCheck);
+      return;
+    }
+
+    switch (deviceOnboardingState?.currentOnboardingStep) {
+      case DeviceOnboardingStep.RestoreSeed:
+      case DeviceOnboardingStep.SafetyWarning:
+      case DeviceOnboardingStep.NewDevice:
+      case DeviceOnboardingStep.NewDeviceConfirming:
+        setStepKey(StepKey.Seed);
+        break;
+      case DeviceOnboardingStep.WelcomeScreen:
+      case DeviceOnboardingStep.SetupChoice:
+        setStepKey(StepKey.Paired);
+        break;
+      case DeviceOnboardingStep.Pin:
+        setStepKey(StepKey.Pin);
+        break;
+      default:
+        break;
+    }
+  }, [deviceOnboardingState]);
+
+  useEffect(() => {
+    if (stepKey >= StepKey.SoftwareCheck) {
+      setStopPolling(true);
+    }
+
+    if (stepKey === StepKey.Ready) {
+      setTimeout(() => setStepKey(StepKey.Exit), readyRedirectDelay / 2);
+    }
+
+    if (stepKey === StepKey.Exit) {
+      setTimeout(handleDeviceReady, readyRedirectDelay / 2);
+    }
+
+    setSteps(
+      defaultSteps.map(step => {
+        const stepStatus =
+          step.key > stepKey ? "inactive" : step.key < stepKey ? "completed" : "active";
+
+        return {
+          ...step,
+          status: stepStatus,
+        };
+      }),
+    );
```

Comment:
Does `steps` need to be a state and set in a useEffect ? A useEffect run after a render, so it would re-trigger a rendering. Maybe you can define `steps` as a normal variable, and set it outside a useEffect, just before the return. `steps` only depends on `stepKey`, which is already being updated by useEffect ;)

Reply:
Also, there is a `<VerticalTimeline steps={steps as any} />` on line 280. Is it possible to not do `as any` ?

Reply:
Done for the `any`, for the `steps` I'd rather avoid it and count on React's optimization rather than compute it at every render

Reply:
Ok ! Good with me :)